src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewModel.swift

```diff
@@ -9,21 +9,33 @@ final class TraceLocationDetailViewModel {
 
 	// MARK: - Init
 	
-	init(_ traceLocation: TraceLocation) {
+	init(_ traceLocation: TraceLocation, eventStore: EventStoringProviding, store: Store) {
+		self.store = store
+		self.eventStore = eventStore
 		self.traceLocation = traceLocation
-		self.locationDescription = traceLocation.description
 		self.locationAddress = traceLocation.address
+		self.locationDescription = traceLocation.description
+		self.shouldSaveToContactJournal = store.shouldAddCheckinToContactDiaryByDefault
 		
+		let duration: Int
 		if let defaultDuration = traceLocation.defaultCheckInLengthInMinutes {
-			self.selectedDurationInMinutes = defaultDuration
+			duration = defaultDuration
 		} else {
 			let eventDuration = Calendar.current.dateComponents(
 				[.minute],
 				from: traceLocation.startDate ?? Date(),
 				to: traceLocation.endDate ?? Date()
 			).minute
 			// the 0 should not be possible since we expect either the defaultCheckInLengthInMinutes or the start and end dates to be available always
-			self.selectedDurationInMinutes = eventDuration ?? 0
+			duration = eventDuration ?? 0
+		}
+		// rounding up to 15
+		let durationStep = 15
+		let reminderMinutes = duration % durationStep
+		if reminderMinutes != 0 {
+			selectedDurationInMinutes = duration + (durationStep - reminderMinutes)
+		} else {
+			selectedDurationInMinutes = duration
 		}
```

Comment:
> We need to do this because the in the creation of the event the time is can any value typed by the user

So why is this not handled during the creation?

Reply:
Good point, i will do so

Reply:
Better solve the real issue than fixing the symptoms ;)